### PR TITLE
Closes #777: Migration for D7 extra info paragraph type.

### LINKED ISSUE #777: Migrate Extra Info paragraph type
See migration mapping doc for details: https://docs.google.com/spreadsheets/d/134UMiYtE8f1YAKYgSukL2JxFzDJF6-TgCAvYFC-vxqQ/edit#gid=0

### PR DESCRIPTION
**Pull request by smitahantzios**
Migration for Paragraph Extra Info to Paragraph Text

## Related Issue

Closes #777 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I've added this Pull Request to the AZ Quickstart Github project and set it to "Review in progress".
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_migration/src/Plugin/migrate/process/ParagraphsExtraInfoFieldMerge.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  public function transform($value, MigrateExecutableInterface $migrate_executable, Row $row, $destination_property) {
+    // Merging the data into one field.
+    $field_uaqs_short_title = $row->getSourceProperty('field_uaqs_short_title');
+    $field_uaqs_body = $row->getSourceProperty('field_uaqs_body');
+    $field_uaqs_link = $row->getSourceProperty('field_uaqs_link');
+    $value['uaqs_text'] = '<div class="border-thick border-top border-azurite">
+    <div class="border card-body">
+       <h3>More information</h3>';
+    if ($field_uaqs_link[0]['url']) {
+      $value['uaqs_text'] .= '<a href="' . $field_uaqs_link[0]['url'] . '" class="{Custom link class}">' . $field_uaqs_link[0]['title'] . '</a>';
```
Comment (joeparsons) [inline modules/custom/az_migration/src/Plugin/migrate/process/ParagraphsExtraInfoFieldMerge.php:30]: The `{Custom link class}` value should actually be the actual field value.
```suggestion
      $value['uaqs_text'] .= '<a href="' . $field_uaqs_link[0]['url'] . '" class="' . $field_uaqs_link[0]['attributes']['class'] . '">' . $field_uaqs_link[0]['title'] . '</a>';
```

**Comment by joeparsons** [comment]
### 2021-06-04 Tasks

- [ ] Merge latest changes from main branch
- [ ] Resolve merge conflicts
- [ ] Use actual field value for custom class (see commit suggestion)

File: modules/custom/az_migration/src/Plugin/migrate/process/ParagraphsExtraInfoFieldMerge.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+   * {@inheritdoc}
+   */
+  public function transform($value, MigrateExecutableInterface $migrate_executable, Row $row, $destination_property) {
+    // Merging the data into one field.
+    $field_uaqs_short_title = $row->getSourceProperty('field_uaqs_short_title');
+    $field_uaqs_body = $row->getSourceProperty('field_uaqs_body');
+    $field_uaqs_link = $row->getSourceProperty('field_uaqs_link');
+    $value['uaqs_text'] = '<div class="border-thick border-top border-azurite">
+    <div class="border card-body">
+       <h3>More information</h3>';
```
Comment (danahertzberg) [inline modules/custom/az_migration/src/Plugin/migrate/process/ParagraphsExtraInfoFieldMerge.php:28]: Should be within the if statement for the link field

File: modules/custom/az_migration/src/Plugin/migrate/process/ParagraphsExtraInfoFieldMerge.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    if ($field_uaqs_link[0]['url']) {
```
Comment (tadean) [inline modules/custom/az_migration/src/Plugin/migrate/process/ParagraphsExtraInfoFieldMerge.php:29]:
```suggestion
    if (!empty($field_uaqs_link[0]['url'])) {
```

File: modules/custom/az_migration/src/Plugin/migrate/process/ParagraphsExtraInfoFieldMerge.php (hunk 2 above)
Comment (smitahantzios) [reply modules/custom/az_migration/src/Plugin/migrate/process/ParagraphsExtraInfoFieldMerge.php]: done

File: modules/custom/az_migration/src/Plugin/migrate/process/ParagraphsExtraInfoFieldMerge.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Drupal\az_migration\Plugin\migrate\process;
+
+use Drupal\migrate\MigrateExecutableInterface;
+use Drupal\migrate\ProcessPluginBase;
+use Drupal\migrate\Row;
+
+/**
+ * Configure Behavior for paragraphs.
```
Comment (tadean) [inline modules/custom/az_migration/src/Plugin/migrate/process/ParagraphsExtraInfoFieldMerge.php:10]: This comment needs updating too I think.

**Review by danahertzberg — APPROVED** [review]
Looks good from the requirements perspective.

**Comment by danahertzberg** [comment]
Fix Troy's suggestion, then ready to merge when all checks pass.

File: modules/custom/az_migration/src/Plugin/migrate/process/ParagraphsExtraInfoFieldMerge.php (hunk 4 above)
Comment (smitahantzios) [reply modules/custom/az_migration/src/Plugin/migrate/process/ParagraphsExtraInfoFieldMerge.php]: Done

**Review by danahertzberg — APPROVED** [review]

**Review by joeparsons — APPROVED** [review]
This one looks good to me now.

**Review by tadean — APPROVED** [review]

**Pull request merged**